packages/viewer/src/components/viewer-pin-group/viewer-pin-group.tsx

```diff
@@ -64,18 +66,47 @@ export class ViewerPinGroup {
   public pinController?: PinController;
 
   /**
-   * Whether or not the pin is "selected"
+   * Whether the pin is "selected"
    */
   @Prop()
   public selected = false;
 
+  /**
+   * @internal
+   * Whether the pin is occluded
+   */
+  @Prop({ mutable: true })
+  public occluded = false;
```

Comment:
I would suggest the following:

* Make this property part of the public API (non-internal), and pass `reflect: true` to the decorator. This way the application can provide it's own styles for occlusion.  
  ```ts
  /**
   * Whether the pin is occluded.
   */
  @Prop({ mutable: true, reflect: true })
  public occluded = false;
  ```
* We're replicating a lot of the behavior that already exists in `<vertex-viewer-dom-element>`. What if we instead exposed a `onOcclusionChanged` event on `<vertex-viewer-dom-element>` that this component listened for and would then set the `occluded` attribute whenever this event is fired?

Reply:
I removed the duplicated logic and set up an event that fires when the occluded attribute is fired.